binaryset/binary_set.c

```diff
@@ -0,0 +1,83 @@
+#include "binary_set.h"
+
+
+void insert(struct BinarySet *set, void *num)
+{
+    struct BinarySetElement *temp = set->element, *prev = set->element;
+ 
+    if (set->element == NULL)
```

Comment:
rvalues should always be on the left hand side of the comparison for security reasons. Eg.
```c
if (NULL == set->element)
```